populus/plugin.py

```diff
@@ -12,17 +24,135 @@
     normalize_object_for_json,
 )
 
+from populus.defaults import (
+    USER_JSON_CONFIG_DEFAULTS,
+    USER_JSON_CONFIG_FILENAME,
+)
+
+from populus.config.helpers import (
+    get_user_default_json_config_file_path,
+)
+
+
+from populus.utils.filesystem import (
+    ensure_path_exists,
+)
 
 CACHE_KEY_MTIME = "populus/project/compiled_contracts_mtime"
 CACHE_KEY_CONTRACTS = "populus/project/compiled_contracts"
 
+BASE_DIR = os.path.dirname(__file__)
+ASSETS_DIR = os.path.join(BASE_DIR, 'assets')
+
+
+testing_project_dir = None
+testing_context = None
+testing_user_config_path = None
+#
+# Hooks
+#
+
+
+def get_populus_option(cmdline_option, ini_option, environ_var, early_config, args):
+
```

Comment:
*whitespace*

Also, can you add a docstring that describes what this function does.

Reply:
ok